core/context.ml

```diff
@@ -0,0 +1,26 @@
+(* Compiler (functional) context/state. *)
+(* TODO localise environment within their associated compilation units. *)
```

Comment:
Maybe it is worth creating a ticket and pointing at it in the comment instead of having multiple "TODO remove" comments?

Reply:
There is alright a ticket for this one. That's #603. This is mostly a comment for me.